### ISSUE TITLE
`Learn.predict` Errors Out if Passed a `PILImage`

### ISSUE THREAD

**Issue by warner-benjamin** [fastai/fastai#3880]
As of the fastai 2.7.11 release, there are reports on the fastai [discord](https://discord.com/channels/689892369998676007/748330524556132364/1075861042875334738) and [forums](https://forums.fast.ai/t/beginner-basics-of-fastai-pytorch-numpy-etc/96285/47) that the [Is it a bird? Creating a model from your own data](https://www.kaggle.com/code/jhoward/is-it-a-bird-creating-a-model-from-your-own-data) Kaggle notebook errors out on the prediction step.

Specifically, here, due to passing in a `PILImage` instead of a Pillow Image or file name:

```python
is_bird,_,probs = learn.predict(PILImage.create('bird.jpg'))
```

Rolling back to fastai 2.7.10 is supposed to [prevent the issue from occurring on Kaggle](https://forums.fast.ai/t/lesson-1-official-topic/95287/469), but when I created a new conda env locally with 2.7.10 I still ran into the error. It's possible that this error occurred due to #3872, or maybe a difference in Pillow versions as I think `PILBase.create` should behave the same in 2.7.10 and 2.7.11 if passed a `PILImage`.

There is a PR to fix the course notebook fastai/course22#52, but I think `learn.predict` should accept a `PILBase` instance, and this issue should be fixed in fastai.

**Comment by nglillywhite** [comment]
Happy to take this one up, will checkout the links and suggest a path forward 😀

**Comment by nglillywhite** [comment]
Hey @warner-benjamin,

I've managed to recreate the error, 2.7.10 works, and 2.7.11 creates this `AttributeError: read`

I'll dig into vision_core and make the changes and tests to cover this.

Is it appropriate to create a predict() test within the vision_learner class and replicate this circumstance despite changes being within the vision_core code? Seems like a cheap way to check/cover a vision learner taking in multiple types but I also don't want to break testing conventions if this isn't the fastai way to test and I should only be testing within the module I've made changes.

**Comment by warner-benjamin** [comment]
@nglillywhite [`vision_learner`](https://github.com/fastai/fastai/blob/master/fastai/vision/learner.py#L209) isn't a class, but rather a method which returns a `Learner` instance.

Assuming the issue is with vision_core, then it would probably suffice to create a test like this current one, except passing a `PILImage` to `PILImage.create`

```python
timg = TensorImage(image2tensor(im))
tpil = PILImage.create(timg)
```
as `Learner.predict` will eventually call `PILImage.create` with whatever was passed to it. (Assuming a standard fastai ImageBlock setup).

**Comment by nglillywhite** [comment]
ok cool, apologies to write 'class', I'm thinking of the notebook/module area and should take better care to use the right terminology, my mistake. Good to know tests within the vision_core section will suffice, thanks for your help 😄

**Comment by nglillywhite** [comment]
https://github.com/fastai/fastai/pull/3884 Opened to fix this one

**Comment by hasnk** [comment]
As a first-time user of the fastai library, I would like to share some observations that I have made. I am unsure if these issues have already been discussed in the forum or in any other issue threads.

I have noticed that when I execute the following code block and then pass it to learn.predict, it raises an `AttributeError: read`

```
dest = PILImage.create(image_cat())
dest.to_thumb(256,256)
```
However, when I execute the following code block and pass it to `learn.predict`, no error is raised:
```
dest = Image.open(image_cat())
dest.to_thumb(256,256)
```
Additionally, I have observed that the type of `dest` differs when I use the two different code blocks. Specifically, when I use the code block:
```
dest = PILImage.create(image_cat())
dest.to_thumb(256,256)
```
`type(dest)` returns `fastai.vision.core.PILImage`, while using the code block:
```
dest = Image.open(image_cat())
dest.to_thumb(256,256)
```
`type(dest)` returns `PIL.JpegImagePlugin.JpegImageFile`

It is worth noting that passing `dest` to `learn.predict` modifies its original type from `PIL.JpegImagePlugin.JpegImageFile` to `fastai.vision.core.PILImage`. 
```
is_cat,_,probs = learn.predict(dest)
print(f"Is this a cat?: {is_cat}.")
print(f"Probability it's a cat: {probs[1].item():.6f}")
```
As a result, even when I execute the aforementioned block twice consecutively, an `AttributeError: read` is raised the second time.
